adal/src/main/java/com/microsoft/aad/adal/WebRequestHandler.java

```diff
@@ -0,0 +1,124 @@
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ * It uses one time async task. WebRequest are wrapped here to prevent multiple
+ * reuses for same tasks. Each request returns a handler for cancel action. Call
+ * this from UI thread to correctly create async task and execute.
+ */
+public class WebRequestHandler implements IWebRequestHandler {
+
+    private static final String TAG = "WebRequestHandler";
+
+    /**
+     * Header for accept.
+     */
+    public static final String HEADER_ACCEPT = AuthenticationConstants.HeaderField.ACCEPT;
+
+    /**
+     * Header for json type.
+     */
+    public static final String HEADER_ACCEPT_JSON = AuthenticationConstants.MediaType.APPLICATION_JSON;
+
+    private UUID mRequestCorrelationId = null;
+
+    private String mCurrentClientVersion = AuthenticationContext.getVersionName();
+
+    private String mBrokerVersion = null;
+
+    @Override
+    public HttpWebResponse sendGet(URL url, Map<String, String> headers) throws IOException {
+        Logger.v(TAG, "WebRequestHandler thread" + android.os.Process.myTid());
+
+        final HttpWebRequest request = new HttpWebRequest(url, HttpWebRequest.REQUEST_METHOD_GET, updateHeaders(headers));
+        return request.send();
+    }
+
+    @Override
+    public HttpWebResponse sendPost(URL url, Map<String, String> headers, byte[] content,
+                                    String contentType) throws IOException {
+        Logger.v(TAG, "WebRequestHandler thread" + android.os.Process.myTid());
+
+        final HttpWebRequest request = new HttpWebRequest(
+                url,
+                HttpWebRequest.REQUEST_METHOD_POST,
+                updateHeaders(headers),
+                content,
+                contentType);
+        return request.send();
+    }
+
+    private Map<String, String> updateHeaders(final Map<String, String> headers) {
+
+        if (mRequestCorrelationId != null) {
+            headers.put(AuthenticationConstants.AAD.CLIENT_REQUEST_ID, mRequestCorrelationId.toString());
+        }
+
+        headers.put(AuthenticationConstants.AAD.ADAL_ID_PLATFORM, "Android");
+        headers.put(AuthenticationConstants.AAD.ADAL_ID_VERSION, mCurrentClientVersion);
+        // Broker Version to report x-client-brkrver
+        if (!StringExtensions.isNullOrBlank(mBrokerVersion)) {
+            headers.put(AuthenticationConstants.AAD.ADAL_BROKER_VERSION, mBrokerVersion);
+        }
+        headers.put(AuthenticationConstants.AAD.ADAL_ID_OS_VER, "" + Build.VERSION.SDK_INT);
+        headers.put(AuthenticationConstants.AAD.ADAL_ID_DM, android.os.Build.MODEL);
+
+        return headers;
+    }
+
+    /**
+     * Sets correlationId.
+     *
+     * @param requestCorrelationId {@link UUID}
+     */
+    public void setRequestCorrelationId(UUID requestCorrelationId) {
+        this.mRequestCorrelationId = requestCorrelationId;
```

Comment:
Since `this` isn't needed, let's remove and make the parameter to this method `final`

Reply:
this is dup to the one in 'common' submodule, I will remove this file.